src/http.cc

```diff
@@ -1461,7 +1465,12 @@ HttpStateData::decodeAndWriteReplyBody()
         addVirginReplyBody(decodedData.content(), decodedData.contentSize());
         if (doneParsing) {
             lastChunk = 1;
-            markParsedVirginReplyAsWhole("http parsed last-chunk");
+            const auto parsedWhole = "http parsed last-chunk";
+            markParsedVirginReplyAsWhole(parsedWhole);
+#if USE_ADAPTATION
+            if (!startedAdaptation)
+#endif
+                fwd->markStoredReplyAsWhole(parsedWhole);
```

Comment:
> is that a bad idea because of how FTP code calls that method?

We cannot do this (while keeping FTP code intact) because FtpRelay calls adaptOrFinalizeReply() after markParsedVirginReplyAsWhole() (fwd->markStoredReplyAsWhole() is conditional on startedAdaptation).